libvast/include/vast/type.hpp

```diff
@@ -1384,6 +1384,13 @@ class record_type final : public stateful_type_base {
 
   /// Returns a new, flattened record type.
   friend record_type flatten(const record_type& type) noexcept;
+
+  /// Enables integration with CAF's type inspection.
+  template <class Inspector>
+  friend auto inspect(Inspector& f, record_type& x) ->
+    typename Inspector::result_type {
+    return f(caf::meta::type_name("vast.record_type"), x.table_);
+  }
```

Comment:
Adding this for concrete types looks rather dangerous to me. CAF's type inspection only works well for semi-regular types, and record type isn't semi-regular, so this will never be usable unless you wrap it into an optional. That means that we have no way to test this inspect handler.

Could you change the code that is using this to use a `type` rather than a `optional<record_type>`?

Reply:
Hm, `optional<record_type>` is the semantically correct type so if we use `type` then we'd have to downcast and assert that it's a record on the call site. Could we instead allow empty `record_type` to make it semi-regular? Or alternatively, we could just create an `inspect(Inspector&, caf::optional<record_type>&)` overload to make sure it only works if the record type is correctly wrapped.

Reply:
We cannot make concrete types semi-regular, that would break quite a few assumptions that we can currently make about record types.

I'm fine with adding an inspect overload like this:

```cpp
template <class Inspector, concrete_type Type>
auto inspect(Inspector& f, std::optional<Type>& x)
    -> typename Inspector::result_type {
  return f(...);
}
```

(Actually not sure if that works generically with concrete type because of the usual ADL wonkiness, but if it doesn't then maybe just for `record_type` as a friend.)